cityofaustin/atd-moped
# Adds DB support for component work types

## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/12229.

The values we keep in the allowed work types  (`moped_component_work_types`) are definitely going to change pending MUG feedback, but I'd like to get this schema established since the access migration depends on it.

## Testing
**URL to test:** Local

**Steps to test:**
Start the hasura cluster. You can test a few queries from graphiql

- Add a work types to an existing project component

```gql
mutation {
  insert_moped_proj_component_work_types_one(
    object: { project_component_id: 2, work_type_id: 1 }
  ) {
    id
  }
}
```

- Query the project component with its work type

```gql
query {
  moped_proj_components_by_pk(project_component_id: 2) {
    project_component_id
    moped_components {
      component_name
    }
    moped_proj_component_work_types {
      id
      moped_work_type {
        name
      }
    }
  }
}
```
- Check out the activity log data

```gql
query {
  moped_activity_log(where: {record_type: {_eq: "moped_proj_component_work_types"}}) {
    activity_id
    record_type
    data: record_data(path: "$.event.data.new")
  }
}
```

- Query from the allowed work types - this data will feed lookup values in a work type autocomplete input

```gql
query {
  moped_component_work_types {
    moped_component {
      component_name
    }
    moped_work_type {
      name
    }
  }
}
````
---
#### Ship list
